### PR TITLE
Add CLI for machine learning classification

### PR DESCRIPTION
**Pull request by alexandreroutier**

### CODE REVIEW

**Comment by mdiazmel** [comment]
Hi @alexandreroutier ,

Seems ok, the CLI is a little bit long but I'm agree that is difficult to propose something better. If the user want to do fine tuning the best thing to do will be to use the modules inside a python script.

Can you please include the cli in the tests: `test/instantation/test_cli.py`?

We'll merge once the doc is completed!

**Comment by alexandreroutier** [comment]
> Hi @alexandreroutier ,
> 
> Seems ok, the CLI is a little bit long but I'm agree that is difficult to propose something better. If the user want to do fine tuning the best thing to do will be to use the modules inside a python script.
> 
> Can you please include the cli in the tests: `test/instantation/test_cli.py`?
> 
> We'll merge once the doc is completed!

I have updated the documentation accordingly to your suggestions.

I'm pinging @nburgos for the review of the documentation.

File: docs/Pipelines/MachineLearning_Classification.md [hunk 1]
```diff
@@ -7,6 +8,9 @@ To build its own classification pipeline, the user can combine three modules bas
 - Algorithm (e.g. support vector machine, logistic regression, random forest)
 - Validation (e.g. K-fold cross validation, repeated K-fold cross validation, repeated hold-out validation)
 
+This combination of modules is wrapped into the `machinelearning-classification` command line interface with default values [[Samper et al., 2018](https://doi.org/10.1016/j.neuroimage.2018.08.042)] for algorithm and validation modules.
+If you want to fine tune these parameters or create your own module(s), please refer to the [Going further](#going-further) section
```
Comment (nburgos) [inline docs/Pipelines/MachineLearning_Classification.md:12]:
```suggestion
If you want to fine tune these parameters or create your own module(s), please refer to the [Going further](#going-further) section.
```
(Full stop missing)

File: docs/Pipelines/MachineLearning_Classification.md [hunk 2]
```diff
@@ -15,48 +19,34 @@ You need to have performed the [`t1-volume`](../T1_Volume) pipeline on your T1-w
...
+- `CAPS_DIRECTORY` is the folder containing the results of the [`t1-volume`](../T1_Volume) and/or the [`pet-volume`](../PET_Volume) pipeline.
+- `GROUP_LABEL` is a string defining the group label for the current analysis, which helps you keep track of different analyses.
+- The third positional argument defines the type of features for classification. It can be:
+    - `RegionBased`: a list of values stored in a TSV file is used as features.
+This list corresponds to PET or T1 image intensities averaged over a set of regions obtained from a brain parcellation when running the [`t1-volume`](../T1_Volume) and/or [`pet-volume`](../PET_Volume) pipeline.
+    - `VoxelBased`: all the voxels of the image are used as features.
+- The fourth positional argument defines the studied modality (`T1w` or `PET`).
+- The fifth positional argument defines the algorithm. It can be:
+    - `DualSVM`: support vector machine (SVM) algorithm
+    - `LogisticRegression`: logistic regression algorithm
```
Comment (nburgos) [inline docs/Pipelines/MachineLearning_Classification.md:43]:
```suggestion
    - `LogisticRegression`: logistic regression algorithm with L2 regularization
```

File: docs/Pipelines/MachineLearning_Classification.md [hunk 3]
```diff
@@ -140,6 +119,65 @@ If `image_type` is `PET`:
...
+### Fine tune algorithm and validation parameters
+
+The `machinelearning-classification` command uses sensible default options (defined in `ml_workflows.py`) that were used for classfication of patients with Alzheimer’s disease [[Samper et al., 2018](https://doi.org/10.1016/j.neuroimage.2018.08.042)].
+
+No matter the combination of modules chosen, the algorithm and validation parameters are:
+
+- `fwhm`: the FWHM value in mm used in the [`t1-volume`](../T1_Volume) and/or the [`pet-volume`](../PET_Volume) pipeline
+- `modulated`: a flag to indicate if, when running the [`t1-volume`](../T1_Volume) pipeline, the image has been modulated or not (`on`, `off`)
+- `use_pvc_data`: use PET data with partial value correction (`True`/`False`).
+By default, PET data with no PVC are used)
```
Comment (nburgos) [inline docs/Pipelines/MachineLearning_Classification.md:133]:
```suggestion
By default, PET data with no PVC are used.
```
(Replace closing bracket by full stop.)

**Comment by nburgos** [comment]
Hi @alexandreroutier,

The documentation seems ok. Any idea why I don't see the modifications on https://aramislab.paris.inria.fr/clinica/docs/public/PR-312? It would be easier to spot formatting problems.

File: docs/Pipelines/MachineLearning_Classification.md [hunk 4]
```diff
@@ -15,48 +19,34 @@ You need to have performed the [`t1-volume`](../T1_Volume) pipeline on your T1-w
...
+    - `VoxelBased`: all the voxels of the image are used as features.
+- The fourth positional argument defines the studied modality (`T1w` or `PET`).
+- The fifth positional argument defines the algorithm. It can be:
+    - `DualSVM`: support vector machine (SVM) algorithm
+    - `LogisticRegression`: logistic regression algorithm
+    - `RandomForest`: random forest algorithm
+- The sixth positional argument defines the validation method. It can be:
+    - `RepeatedHoldOut`: repeated hold-out validation
+    - `RepeatedKFoldCV`: repeated K-fold cross validation
+- `SUBJECTS_VISITS_TSV` is a TSV file containing the `participant_id` and the `session_id` columns
```
Comment (nburgos) [inline docs/Pipelines/MachineLearning_Classification.md:48]:
```suggestion
- `SUBJECTS_VISITS_TSV` is a TSV file containing the `participant_id` and the `session_id` columns.
```

File: docs/Pipelines/MachineLearning_Classification.md [hunk 5]
```diff
@@ -69,29 +59,18 @@ sub-CLNC0004      AD
...
-- `precomputed_kernel`: to load the precomputed kernel if it exists
-- `mask_zeros`: a flag to indicate if zero-valued voxels should be taken into account for the classification (`True`/`False`)
-- `n_iterations`: number of times a task is repeated
-- `grid_search_folds`: number of folds to use for the hyper-parameter grid search (e.g. 10)
-- `c_range`: range used to select the best value for the C parameter, in the logspace
-- `n_threads`: number of threads used if run in parallel
-- `test_size`: percentage (between 0 and 1) representing the size of the test set for each shuffle split
-- `balanced`:  option to balance the weights according to the number of samples
-- `penalty`: type of penalty ("l2" or "l1")
+- `OUTPUT_DIRECTORY`: the directory where outputs are saved
```
Comment (nburgos) [inline docs/Pipelines/MachineLearning_Classification.md:62]:
```suggestion
- `OUTPUT_DIRECTORY`: the directory where outputs are saved.
```

**Comment by alexandreroutier** [comment]
Hi @mdiazmel,

@nburgos reviewed the documentation. If the code suits you, I think it would be cleaner if this PR is Squash & Merge.

Alex

**Comment by mdiazmel** [comment]
LGTM,
I'll merge this one!

**Pull request merged**